pkg/kv/kvserver/protectedts/ptstorage/storage.go

```diff
@@ -225,18 +324,47 @@ func (p *storage) GetState(ctx context.Context, txn *kv.Txn) (ptpb.State, error)
 	}, nil
 }
 
+func (p *storage) deprecatedGetRecords(ctx context.Context, txn *kv.Txn) ([]ptpb.Record, error) {
+	it, err := p.ex.QueryIteratorEx(ctx, "protectedts-deprecated-GetRecords", txn,
+		sessiondata.InternalExecutorOverride{User: security.NodeUserName()},
+		getRecordsWithoutTargetQuery)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to read records")
+	}
+
+	var ok bool
+	var records []ptpb.Record
+	for ok, err = it.Next(ctx); ok; ok, err = it.Next(ctx) {
+		var record ptpb.Record
+		if err := rowToRecord(ctx, it.Cur(), &record, p.settings, p.knobs); err != nil {
+			log.Errorf(ctx, "failed to parse row as record: %v", err)
+		}
+		records = append(records, record)
+	}
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to read records")
+	}
+	return records, nil
+}
+
 func (p *storage) getRecords(ctx context.Context, txn *kv.Txn) ([]ptpb.Record, error) {
+	if !p.settings.Version.IsActive(ctx, clusterversion.AlterSystemProtectedTimestampAddColumn) ||
+		!p.knobs.EnableProtectedTimestampForMultiTenant {
+		return p.deprecatedGetRecords(ctx, txn)
+	}
+
 	it, err := p.ex.QueryIteratorEx(ctx, "protectedts-GetRecords", txn,
 		sessiondata.InternalExecutorOverride{User: security.NodeUserName()},
-		getRecordsQuery)
+		getRecordsWithoutTargetQuery)
```

Comment:
this is wrong, it should be getRecordsQuery. I'm just waiting for CI to shake out all failures before repushing.

Reply:
done.